service/src/main/java/org/cbioportal/service/impl/MolecularProfileCorrelationServiceImpl.java

```diff
@@ -0,0 +1,589 @@
+package org.cbioportal.service.impl;
```

Comment:
@oplantalech @pieterlukasse When I look at this class, I see a lot of very similar code blocks repeating over and over again. There must be a better way to do this without so many repetition.

Reply:
@ersinciftci I've been working to remove the repetitive code. I think that looks much better now 😃